src/assocPath.js

```diff
@@ -41,7 +43,8 @@ module.exports = _curry3(function assocPath(path, val, obj) {
   }
   if (_isInteger(idx) && _isArray(obj)) {
     var arr = [].concat(obj);
-    arr[idx] = val;
+    var arrIdx = idx < 0 ? arr.length + idx : idx;
```

Comment:
I think it is the other way around? `assoc` can be easily defined in terms of `assocPath`. They have historically had divergent behavior with certain input, which is why this hasn't happened yet. Maybe it is time we do that, as this PR as it stands would be yet another difference.